### PR TITLE
Optical physicslist tool

### PR DESCRIPTION
**Pull request by atolosadelgado**
Hi,

Optical physics will be needed for optical calorimeters and the ARC. 

I have copied the tool which allows to register the Optical Physics on top of a full physics list. 
https://github.com/HEP-FCC/dual-readout/blob/master/DRsim/DRsimG4Full/src/components/SimG4OpticalPhysicsList.cpp

I have added setters for the 3 optical processes: Cerenkov, Scintillation and Transition Radiation. 

Cerenkov and Scintillation are enabled by default, as in the original file.

The code is rather small, so a test for just this is not worth it. However, it might be healthy for the repository to have a minimal simulation as test, to notice changes in Geant4 or DD4hep.

Best,
Alvaro

### CODE REVIEW

File: SimG4Components/src/SimG4OpticalPhysicsList.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+public:
+  explicit SimG4OpticalPhysicsList(const std::string& aType, const std::string& aName, const IInterface* aParent);
+  virtual ~SimG4OpticalPhysicsList();
+
+  virtual StatusCode initialize();
+  virtual StatusCode finalize();
+  virtual G4VModularPhysicsList* physicsList();
+  Gaudi::Property<bool> SetCerenkov{this, "SetCerenkov", true, "Bool variable that enables Cerenkov process. Default true."};
+  Gaudi::Property<bool> SetScintillation{this, "SetScintillation", true, "Bool variable that enables Scintillation process.. Default true."};
+  Gaudi::Property<bool> SetTransitionRadiation{this, "SetTransitionRadiation", false, "Bool variable that enables transition_radiation process. . Default false."};
```
Comment (BrieucF) [inline SimG4Components/src/SimG4OpticalPhysicsList.h:21]:
```suggestion
  Gaudi::Property<bool> SetTransitionRadiation{this, "SetTransitionRadiation", false, "Bool variable that enables transition_radiation process. Default false."};
```

File: SimG4Components/src/SimG4OpticalPhysicsList.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class SimG4OpticalPhysicsList : public AlgTool, virtual public ISimG4PhysicsList {
+public:
+  explicit SimG4OpticalPhysicsList(const std::string& aType, const std::string& aName, const IInterface* aParent);
+  virtual ~SimG4OpticalPhysicsList();
+
+  virtual StatusCode initialize();
+  virtual StatusCode finalize();
+  virtual G4VModularPhysicsList* physicsList();
+  Gaudi::Property<bool> SetCerenkov{this, "SetCerenkov", true, "Bool variable that enables Cerenkov process. Default true."};
+  Gaudi::Property<bool> SetScintillation{this, "SetScintillation", true, "Bool variable that enables Scintillation process.. Default true."};
```
Comment (BrieucF) [inline SimG4Components/src/SimG4OpticalPhysicsList.h:20]:
```suggestion
  Gaudi::Property<bool> SetScintillation{this, "SetScintillation", true, "Bool variable that enables Scintillation process. Default true."};
```

File: SimG4Components/src/SimG4OpticalPhysicsList.h [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#ifndef SimG4OpticalPhysicsList_h
+#define SimG4OpticalPhysicsList_h 1
+
+// Gaudi
+#include "GaudiKernel/AlgTool.h"
+#include "GaudiKernel/ToolHandle.h"
+
+// FCCSW
+#include "k4Interface/ISimG4PhysicsList.h"
+
```
Comment (BrieucF) [inline SimG4Components/src/SimG4OpticalPhysicsList.h:10]: Hi Alvaro, can you add a brief description og what the tool does? Like e.g. in https://github.com/HEP-FCC/k4SimGeant4/blob/main/SimG4Components/src/SimG4SaveCalHits.h#L20-L30

File: SimG4Components/src/SimG4OpticalPhysicsList.h [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#ifndef SimG4OpticalPhysicsList_h
+#define SimG4OpticalPhysicsList_h 1
+
+// Gaudi
+#include "GaudiKernel/AlgTool.h"
+#include "GaudiKernel/ToolHandle.h"
+
+// FCCSW
```
Comment (andresailer) [inline SimG4Components/src/SimG4OpticalPhysicsList.h:8]: Actually this is the k4FWCore location https://github.com/key4hep/k4FWCore/blob/main/k4Interface/include/k4Interface/ISimG4PhysicsList.h

```suggestion
// k4FWCore
```

But there is also the same file here https://github.com/HEP-FCC/k4SimGeant4/blob/main/SimG4Interface/include/SimG4Interface/ISimG4PhysicsList.h
And it should be dropped in k4SimGeant4.

Comment (atolosadelgado) [reply SimG4Components/src/SimG4OpticalPhysicsList.h:8]: I took as template another physics lists existing in this repo. If refactoring of this and other physics lists tools is needed, maybe it is worth to do it in a separate PR?

Comment (andresailer) [reply SimG4Components/src/SimG4OpticalPhysicsList.h:8]: Yes, probably worth doing a separate PR.

**Pull request merged**